src/netaddress.cpp

```diff
@@ -996,8 +996,25 @@ CSubNet::CSubNet():
 
 CSubNet::CSubNet(const CNetAddr& addr, uint8_t mask) : CSubNet()
 {
-    valid = (addr.IsIPv4() && mask <= ADDR_IPV4_SIZE * 8) ||
-            (addr.IsIPv6() && mask <= ADDR_IPV6_SIZE * 8);
+    switch (addr.m_net) {
+    case NET_IPV4:
+        valid = mask <= ADDR_IPV4_SIZE * 8;
+        break;
+    case NET_IPV6:
+        valid = mask <= ADDR_IPV6_SIZE * 8;
+        break;
+    case NET_ONION:
+    case NET_I2P:
+    case NET_CJDNS:
+        valid = true;
```

Comment:
We shouldn't just *ignore* `mask`... Might be best to only change the single-address form?

Reply:
Hm, I don't know if we need to check for the various cases here.
```cpp
/// Size of TORv2 address (in bytes).
static constexpr size_t ADDR_TORV2_SIZE = 10;

/// Size of TORv3 address (in bytes). This is the length of just the address
/// as used in BIP155, without the checksum and the version byte.
static constexpr size_t ADDR_TORV3_SIZE = 32;

/// Size of I2P address (in bytes).
static constexpr size_t ADDR_I2P_SIZE = 32;

/// Size of CJDNS address (in bytes).
static constexpr size_t ADDR_CJDNS_SIZE = 16;
```

Reply:
Done: https://github.com/bitcoin/bitcoin/pull/20852#issuecomment-756080044